src/maxdiffusion/generate_sdxl.py

```diff
@@ -132,14 +129,28 @@ def run(config):
     mesh=mesh
   )
 
+  # if this checkpoint was trained with maxdiffusion
+  # the training scheduler was saved with it, switch it
+  # to a Euler scheduler
+  if isinstance(pipeline.scheduler, FlaxDDPMScheduler):
```

Comment:
how does this check if the checkpoint is loaded from `maxdiffusion based training`?

Reply:
DDPM scheduler is used for training and it should not be used for inference as it will result in noisy images. More broadly this ensures that someone doesn't accidentally use the wrong scheduler.